lib/phoenix/transports/long_poller/server.ex

```diff
@@ -81,19 +83,27 @@ defmodule Phoenix.Transports.LongPoller.Server do
       {:error, reason} ->
         :ok = broadcast_from(state, {:error, :dispatch, reason, ref})
         {:noreply, state}
-      :ignore ->
-        :ok = broadcast_from(state, {:error, :dispatch, :ignore, ref})
-        {:noreply, state}
     end
   end
 
   @doc """
   Forwards replied/broadcasted message from Channels back to client.
   """
-  def handle_info({:socket_push, msg}, state) do
+  def handle_info(%Message{} = msg, state) do
     publish_reply(msg, state)
   end
 
+  def handle_info(%Reply{} = reply, state) do
+    %{topic: topic, status: status, payload: payload, ref: ref} = reply
+
+    # TODO: Don't duplicate the reference. The client
```

Comment:
@chrismccord I will need your help for this one. We are encoding the reference in the payload and in the message. We don't need the payload one but we need to change the client for that.